src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -1761,13 +1761,17 @@ def binary_op(
             # -------------------------
             return self._binary_op_scalar_rhs(op, other, fill_value)
 
-        if not isinstance(other, (Series, DataFrame)) and is_list_like(other):
+        if not isinstance(other, (Series, DataFrame)):
```

Comment:
let's make `is_list_like` work for the new index.

Reply:
Fixed